Product/Production/Common/CONNECTCoreLib/src/main/java/gov/hhs/fha/nhinc/opensaml/extraction/OpenSAMLAssertionExtractorImpl.java

```diff
@@ -564,14 +573,17 @@ private static AssertionType initializeAssertion() {
      * @param target target assertion
      */
     private static void populateSubjectRole(final Attribute attribute, final AssertionType target) {
-        LOG.debug("Executing Saml2AssertionExtractor.populateSubjectRole...");
+        LOG.trace("Executing Saml2AssertionExtractor.populateSubjectRole...");
 
-        XMLObject subjRoleAttribute = attribute.getAttributeValues().get(0);
-        XMLObject roleElement = subjRoleAttribute.getOrderedChildren().get(0);
+        if (!CollectionUtils.isEmpty(attribute.getAttributeValues()) && attribute.getAttributeValues().get(0) != null
```

Comment:
Suggestion: Extract into private method to reduce duplicate logic in populatePurposeOfUseAttribute.